### PR TITLE
Move three windings transformers, dangling lines and voltage levels with unknown buses in network area diagram

### PR DESCRIPTION
**Pull request by massimo-ferraro**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
no

**What kind of change does this PR introduce?**
feature

**What is the current behavior?**
Three windings transformers, dangling lines and voltage levels with unknown buses are not correctly moved

**What is the new behavior (if this is a feature change)?**
Three windings transformers, dangling lines and voltage levels with unknown buses can be moved

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

### CODE REVIEW

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 1]
```diff
@@ -563,14 +608,35 @@ export class NetworkAreaDiagramViewer {
...
+            edgeType == DiagramUtils.EdgeType.DANGLING_LINE &&
+            edgeNodes[1] != null
+        ) {
+            this.redrawBoundaryNode(
+                edgeNodes[1],
+                DiagramUtils.getAngle(edgeStart2, edgeMiddle),
+                nodeRadius2[1]
+            );
+            if (this.selectedElement?.id == edgeNodes[1].id) {
+                boundaryNodeMoved = true;
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:626]: I'd find it clearer to call directly `redrawOtherVoltageLevelNode` here. Then you can get rid of the `boundaryNodeMoved` boolean
```suggestion
                this.redrawOtherVoltageLevelNode(edgeNodes[0], [edge]);
```

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 2]
```diff
@@ -563,14 +608,35 @@ export class NetworkAreaDiagramViewer {
...
+            }
+        }
+        if (
+            edgeType != DiagramUtils.EdgeType.DANGLING_LINE ||
+            boundaryNodeMoved
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:631]: can you transform this into a else to have a clear separation danglingLine/not danglingLine? this probably means removing `edgeNodes[1] != null` in the previous if condition

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 3]
```diff
@@ -1024,6 +1108,9 @@ export class NetworkAreaDiagramViewer {
         traversingBusEdgesAngles.push(
             traversingBusEdgesAngles[0] + 2 * Math.PI
         );
+        traversingBusEdgesAngles.sort(function (a, b) {
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:1111]: I don't get why this is needed, the angles are all computed with atan2, so are between -pi and pi. The last angle added is therefore between pi and 3*pi, the second sorting shouldn't change anything. What am I missing?

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 4]
```diff
@@ -1114,4 +1201,94 @@ export class NetworkAreaDiagramViewer {
             this.redrawVoltageLevelNode(otherNode, busNodeEdges, movedEdges);
         }
     }
+
+    private moveThreeWtEdge(
+        edge: SVGGraphicsElement,
+        edgeNode: SVGGraphicsElement,
+        offset: Point
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:1208]: mousePosition

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 5]
```diff
@@ -1114,4 +1201,94 @@ export class NetworkAreaDiagramViewer {
...
+                        busNodeId1 != null ? +busNodeId1 : -1
+                    );
+                    const edgeStart = DiagramUtils.getDistance(
+                        DiagramUtils.getPosition(edgeNodes[0]),
+                        DiagramUtils.getPosition(edgeNodes[1]),
+                        unknownBusNode1
+                            ? nodeRadius1[1] +
+                                  this.svgParameters.getUnknownBusNodeExtraRadius()
+                            : nodeRadius1[1]
+                    );
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:1233]: duplicate code with moveStraightEdge content. We could create a function to factorize this, and use it also for both sides of moveStraightEdge?

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts [hunk 6]
```diff
@@ -1114,4 +1201,94 @@ export class NetworkAreaDiagramViewer {
...
+                        ',' +
+                        edgeStart.y.toFixed(2) +
+                        ' ' +
+                        edgeEnd.x.toFixed(2) +
+                        ',' +
+                        edgeEnd.y.toFixed(2);
+                    twtEdge.setAttribute('points', polylinePoints);
+                    // move edge arrow and label
+                    if (edgeNode.children.length > 1) {
+                        this.moveEdgeArroWAndLabel(
```
Comment (flo-dup) [inline src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts:1256]: just realized the typo: 

```suggestion
                        this.moveEdgeArrowAndLabel(
```

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts (hunk 1 above)
Comment (massimo-ferraro) [reply src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts]: done

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts (hunk 2 above)
Comment (massimo-ferraro) [reply src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts]: done

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts (hunk 3 above)
Comment (massimo-ferraro) [reply src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts]: This second sorting was added to correctly handle the redrawing after some node moving. Is seems that changing the addition of the last angle to the angle list would make this second sorting useless (so it has been removed).

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts (hunk 5 above)
Comment (massimo-ferraro) [reply src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts]: Added method getEdgeStart

File: src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts (hunk 6 above)
Comment (massimo-ferraro) [reply src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts]: Fixed

**Review by flo-dup — APPROVED** [review]

**Pull request merged**